tensorflow/core/tpu/tpu_initializer_helper.cc

```diff
@@ -99,11 +99,11 @@ bool IsTpuUsed(int64_t pid) {
 // processes owned by another user.
 // TODO (shahrokhi) use tensorflow/core/platform/filesystem (GetChildren) for
 // this.
-bool FindAndLogLibtpuProcess() {
+std::string FindAndLogLibtpuProcess() {
```

Comment:
Let's refactor this a bit. Let's make this function:
`StatusOr<int> FindLibtpuProcess()`
So instead of creating the whole error message, it'll just try to find the PID and return that. If it can't find it for some reason, you can put a short reason in the status (e.g. "Can't open /proc"), but we won't actually use the error status.

Then in TryAcquireTpuLock, call `FindLibtpuProcess()`, and if it succeeds:
`return Status(absl::StrCat("libtpu.so is already in use by process with pid ", pid, ...))`
And if it failed to get the PID:
`return Status(libtpu.so already in use by another process probably owned...")`

This way all the error-message-generating logic is in TryAcquireTpuLock, instead of splitting across there and here.